### DISCUSSION TITLE
K-Medoids in R Studio

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#692] [Q&A] [answered]
Saya punya data seperti ini : 
<img width="612" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/1c5259bb-e96f-4ba6-ae48-008e04c071fc">
Dengan unique IPnya sebanyak 323
<img width="432" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/c055a98f-f851-49fc-b464-040213788de2">

Saya ingin melakukan clustering dengan menggunakan metode k-medoids, namun hasilnya seperti ini : 
<img width="618" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/4360a38a-9f00-4210-aa84-41528ef6522b">

IPnya malah tinggal 6
<img width="356" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/ce5caca8-5d4f-4416-abaf-e32af03b4d67">

Saya ingin mengelompokkan IP dengan behavior yang mirip ke dalam tiap-tiap cluster.
Mohon bantuannya kak, terima kasih 🙏

**Comment by kinantytasyao** [comment]
Dear @yohanespm77 ,

Jika ingin mengelompokkan source_ip maka sebaiknya Bapak tidak memasukkan source_ip sebagai salah satu variabel untuk clusteringnya, melainkan menjadikan source_ip sebagai index baris (yang dilakukan menggunakan fungsi `rownames()` di R). Nantinya ketika sudah selesai melakukan clustering Bapak bisa mengecek tiap observasi baris source_ip masuk ke cluster mana menggunakan `$cluster` 

Demikian yang dapat kami sampaikan. Semoga membantu.

**Reply by yohanespm77** [reply]
Dear mba @kinantytasyao 

Sebelumnya saya udah coba untuk mejadikan source_ip sebagai index kak, namun gagal sebab source_ipnya banyak yang berulang.

<img width="388" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/10dbd91a-a812-4b41-b352-743f9e1fec10">

**Reply by kinantytasyao** [reply]
Kalau begitu bisa diambil data yang nilai source_ip nya unik saja Pak, bisa dengan cara mengaplikasikan fungsi `match()` pada data kita seperti yang dipelajari di kelas DV untuk mengambil data yang nilainya unik (tidak berulang) dari data awal kita

**Reply by yohanespm77** [reply]
Tapi saya butuh untuk mengambil semua datanya mba, untuk melihat behaviournya berdasarkan frekuensinya juga. 

Data versi lengkapnya seperti ini : 
<img width="608" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/5e6e8292-e69b-45b8-8583-18fde86cc856">

Kalau diambil nilai uniquenya saja ada informasi yang kurang. Contohnya, satu IP bisa melakukan lebih dari satu status_code. Kalau diambil 1 saja IPnya (unique), status_code yang lain pada IP tsb jadi tidak digunakan.

**Reply by ELITSZORD** [reply]
@yohanespm77 Bisa gk pak kalau bikin beberapa kolom baru dimana misal kolom pertama untuk mendeskripsikan jumlah berpa kali muncul nya status code 0-200 per ip addressnya (supaya nanti bisa menjadi unique identifier), kemudian kolom berikutnya jumlah berapa kali munculnya status code 201-400 per ip addressnya, dan seterusnya. Sehingga si IP address nya bisa sebagai unique identifier dimana bisa mewakili beberapa kolom yang dapat merepresentasikan behaviour/pola dari suatu 1 data IP address.

**Reply by kevbow** [reply, marked as answer]
Halo @yohanespm77, untuk menjaga agar forum diskusi tetap berjalan kondusif, kami izin melakukan lock/close diskusi ini karena belum ada respon setelah 7 hari respon terakhir. Semoga dapat dipahami.
